### ISSUE TITLE
Allow arguments to be passed to the JVM

### ISSUE THREAD

**Issue by jamessimone** [dangmai/prettier-plugin-apex#356]
I'm attempting to get the best of both worlds - Java 11 _and_ the usage of ZGC to reduce memory consumption. The sfdx plugins are a huge memory drain at the moment, particularly on larger projects. Unfortunately, it doesn't seem like `prettier-plugin-apex` supports the passing of `JAVA_OPTS` to the JVM:

```
["ERROR" - 1:51:42 PM] Error formatting document.
Error: Error occurred during initialization of VM
Option -XX:+UseZGC not supported
```

I'm running AdoptOpenJDK 11 with the following set as my `JAVA_OPTS` (and `JAVA_HOME` set to the JDK's location):

```
-XX:+UnlockExperimentalVMOptions -XX:+UseZGC
```

In looking at the contents of `vendor/apex-ast-serializer/bin` it certainly seems like everything should be working correctly - and indeed, the SFDX extensions run with reduced memory consumption as one would expect. If I can get `prettier-plugin-apex` working correctly with these JVM options, I'll be set. Any tips? Thank you!

**Comment by dangmai** [comment]
Hi, thank you for opening the bug report. To be completely honest with you, I have not worked in the JVM ecosystem in a professional context for a long time, so I am just relying on the tooling to "do the right thing" for me. The Java part of this program is [apex-ast-serializer](https://github.com/dangmai/apex-ast-serializer), and the binaries are created by the Gradle plugin `application` - you can see the Gradle source code [here](https://github.com/dangmai/apex-ast-serializer/blob/master/build.gradle). I am happy to merge in any changes that you can propose to make your workflow work, but I don't really know how to get there myself.

A few things that we can do to debug this and get somewhere:

- If you pass in other types of JVM arguments (e.g. max heap size), does it work correctly with Prettier Apex?
- If none of the JVM argument works, then maybe we do need to some argument to Prettier Apex that can be passed through to apex-ast-serializer when we invoke the binaries.
- Another thought: if you are trying to optimize performance, there is always an option of running apex-ast-serializer as a standalone HTTP server, maybe you can simply just invoke the binaries yourself and pass in whatever JAVA_OPTS you want?

**Comment by jamessimone** [comment]
@dangmai thanks for the speedy response. To be honest, it’s 5+ years since I worked with Java itself.

 I’ll let you know if any other JVM_OPTS work out of the box; I’m also open to the pass-through plugin option as well. I’ll report back tomorrow with my findings. Thanks for making this plugin by the way - I used to spend way too much time formatting code, and this plugin has truly helped to standardize writing Apex in a super helpful way.
